add script to compare results and fail at regression

laithsakka opened the pull request:
in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132215
* __->__ #132099
* #131751
* #131475



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

laithsakka committed:
add script to compare results and fail at regression

[ghstack-poisoned]

laithsakka committed:
Update on "add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on "[draft] add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 commented on benchmarks/dynamo/pr_time_benchmarks/compare_results.py:
```diff
@@ -0,0 +1,55 @@
+                print(f"{benchmark_name}, {metric}, failed")
+                fail = True
+
+    if fail:
+        sys.exit(1)
+    else:
+        print("all benchmarks passed")
```
Might be good to phase this out in a two separate PRs, where the second PR actually turns it ON. I expect some level of CI instability, which will cause reverts. In such case, only 2nd PR will be reverted.

laithsakka committed:
Update on " add script to compare results and fail at regression"

in case of success those are the logs
```
looking for regressions:
+ python benchmarks/dynamo/pr_time_benchmarks/compare_results.py /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_before.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_benchmarks_after.txt /var/lib/jenkins/workspace/test/test-reports/pr_time_compare_results.txt
update_hint_regression, instruction_count, -0.35%
all benchmarks passed
```
in case of failure check
 https://github.com/pytorch/pytorch/pull/132215



cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
